### PR TITLE
Implement Window API

### PR DESCRIPTION
**Pull request by simon816**
**SpongeCommon** | [SpongeAPI](https://github.com/SpongePowered/SpongeAPI/pull/1189)

Supersedes #663 (didn't realize force-push will not allow re-opening PRs)

### CODE REVIEW

File: src/main/java/org/spongepowered/common/gui/window/AbstractSpongeBlockContainerWindow.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import org.spongepowered.api.gui.window.LocationBackedWindow;
+import org.spongepowered.api.world.Location;
+import org.spongepowered.api.world.World;
+import org.spongepowered.common.util.VecHelper;
+
+import java.util.Optional;
+
+public abstract class AbstractSpongeBlockContainerWindow extends AbstractSpongeContainerWindow implements LocationBackedWindow {
+
+    protected Location<World> location;
```
Comment (liach) [inline src/main/java/org/spongepowered/common/gui/window/AbstractSpongeBlockContainerWindow.java:38]: `@Nullable`?

File: src/main/java/org/spongepowered/common/gui/window/AbstractSpongeContainerWindow.java [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+import net.minecraft.network.play.server.S2DPacketOpenWindow;
+import net.minecraft.world.IInteractionObject;
+import org.spongepowered.api.gui.window.ContainerWindow;
+import org.spongepowered.common.interfaces.entity.player.IMixinEntityPlayerMP;
+
+import java.util.Optional;
+
+public abstract class AbstractSpongeContainerWindow extends AbstractSpongeWindow implements ContainerWindow {
+
+    protected abstract IInteractionObject provideInteractionObject();
```
Comment (kashike) [inline src/main/java/org/spongepowered/common/gui/window/AbstractSpongeContainerWindow.java:40]: `@Nullable`

File: src/main/java/org/spongepowered/common/gui/window/SpongeHorseInventoryWindow.java [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+import org.spongepowered.api.data.manipulator.mutable.entity.HorseData;
+import org.spongepowered.api.entity.living.animal.Horse;
+import org.spongepowered.api.gui.window.HorseInventoryWindow;
+import org.spongepowered.common.interfaces.entity.player.IMixinEntityPlayerMP;
+
+import java.util.Optional;
+
+public class SpongeHorseInventoryWindow extends AbstractSpongeContainerWindow implements HorseInventoryWindow {
+
+    private EntityHorse horse;
```
Comment (kashike) [inline src/main/java/org/spongepowered/common/gui/window/SpongeHorseInventoryWindow.java:46]: `@Nullable`

File: src/main/java/org/spongepowered/common/gui/window/SpongeHorseInventoryWindow.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+    private HorseData horseData;
```
Comment (liach) [inline src/main/java/org/spongepowered/common/gui/window/SpongeHorseInventoryWindow.java:47]: `@Nullable`

**Comment by gabizou** [comment]
Please update the target branch to bleeding (Edit this PR and change the target branch) and reopen.

**Comment by Zidane** [comment]
Closing this given the age that has passed and the lack of activity. If someone wants to continue the work done here then we can re-open at a later time.

**Pull request closed without merging**